plugin/trino-thrift-api/src/main/java/io/trino/plugin/thrift/api/datatypes/TrinoThriftBigintArray.java

```diff
@@ -144,12 +145,18 @@ public String toString()
 
     public static TrinoThriftBlock fromBlock(Block block)
     {
-        checkArgument(block instanceof AbstractArrayBlock, "block is not of an array type");
-        AbstractArrayBlock arrayBlock = (AbstractArrayBlock) block;
-        int positions = arrayBlock.getPositionCount();
+        int positions = block.getPositionCount();
         if (positions == 0) {
             return bigintArrayData(new TrinoThriftBigintArray(null, null, null));
         }
+        if (block instanceof RunLengthEncodedBlock && block.isNull(0)) {
+            boolean[] nulls = new boolean[positions];
+            Arrays.fill(nulls, true);
+            return bigintArrayData(new TrinoThriftBigintArray(nulls, null, null));
```

Comment:
Can you add a test?

This would fail
```
checkArgument(totalSize(nulls, sizes) == numberOfValues(values), "total number of values doesn't match expected size");
```
in `TrinoThriftBigintArray`

Reply:
> Can you add a test?

`TestReadWrite.testSingleRowPageReadWrite` already tests null RLE 



> This would fail
> 
> ```
> checkArgument(totalSize(nulls, sizes) == numberOfValues(values), "total number of values doesn't match expected size");
> ```
> 
> in `TrinoThriftBigintArray`

WDYM? can this fail when `nulls` is all true and `sizes` and `values` are both null?

Reply:
Yes. It would fail. Please inspect code. I think test is lacking

Reply:
> Yes. It would fail. Please inspect code (mentioned assertion). I think test is lacking


Could you be more specific? I think my code inspection skills are lacking..
My _inspection_ is this: `numberOfValues(values)` always returns 0 for `null` `values`.
`totalSize` also returns 0 if all nulls are set to true because this condition is always false `if (nulls == null || !nulls[i]) {`

Reply:
Ah. You are right